### DISCUSSION TITLE
The simplest CMakeLists.txt for a Linux project

### DISCUSSION THREAD

**Discussion by pauljurczak** [ispc/ispc#2724] [Q&A] [answered]
I have a very limited knowledge of CMake. I'm looking for a standalone CMakeLists.txt file with modern ISPC support, which can built the example project from https://github.com/ispc/ispc/tree/main/examples/cpu/simple. So far I have the C++ part:
```
cmake_minimum_required(VERSION 3.22)
project(ispc LANGUAGES C CXX)

find_program(ISPC_EXECUTABLE ispc)
find_package(fmt REQUIRED)

include_directories(
  /home/paul/include
)

add_executable(ispc simple.cpp)

add_compile_options(-O3 -Wno-narrowing -Wall -Wextra -pedantic -Werror -Wdeprecated-enum-enum-conversion -v -ffast-math)

target_precompile_headers(ispc PRIVATE 
  /home/paul/include/common.hpp
)

target_link_libraries (ispc
  fmt::fmt
)

set(CMAKE_BUILD_TYPE "Release")
set(CMAKE_VERBOSE_MAKEFILE TRUE)
target_compile_features(ispc PUBLIC cxx_std_23)
message(STATUS "C++ compiler: ${CMAKE_CXX_COMPILER_ID} v${CMAKE_CXX_COMPILER_VERSION}")
```
I don't know how to generate the required `simple_ispc.h` header and the rest of ispc dependencies. I searched around and couldn't find the boilerplate for a modern CMakeLists.txt file.

**Comment by mikemccarty-vertex** [comment]
I'm not a cmake expert either but at minimum, I think you need:
```
project(ispc LANGUAGES C CXX ISPC)
```
I'm not sure the `find_program()` is necessary.

**Comment by nurmukhametov** [comment, marked as answer]
There are a few good examples [here](https://gitlab.kitware.com/cmake/cmake/-/tree/master/Tests/ISPC).

**Reply by pauljurczak** [reply]
Thank you. I think https://gitlab.kitware.com/cmake/cmake/-/blob/master/Tests/ISPC/TryCompile/CMakeLists.txt is what I was looking for.
